presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -107,5 +111,29 @@ public Expression rewriteExtract(Extract node, Void context, ExpressionTreeRewri
 
             throw new UnsupportedOperationException("not yet implemented: " + node.getField());
         }
+
+        @Override
+        public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            if (!REWRITE_ENABLED_FUNCTIONS.contains(node.getName())) {
+                return null;
```

Comment:
Is returning `null` the expected behavior, or should we return the original expression of the node? 

I did not find `null` handling in this class and the caller of `canonicalizeExpression`. I might be over-alarming since all the tests are passing.

Reply:
Yes returning null if not rewriting the expression. Handled here: https://github.com/prestodb/presto/blob/master/presto-parser/src/main/java/com/facebook/presto/sql/tree/ExpressionTreeRewriter.java#L480